### PR TITLE
`order` option for `copy`

### PR DESCRIPTION
**Pull request by unnonouno**
I supported order option in `copy` method using `astype`.
merge #111 first

### CODE REVIEW

File: cupy/core/core.pyx [hunk 1]
```diff
@@ -342,38 +342,32 @@ cdef class ndarray:
...
+        with self.device:
+            x = self.astype(self.dtype, copy=True, order=order)
+
+        if device.Device() != self.device:
+            newarray = ndarray(x.shape, dtype=x.dtype)
+            newarray._set_shape_and_strides(x._shape, x._strides)
+            newarray.data.copy_from_device(x.data, x.nbytes)
+            return newarray
+        else:
+            return x
```
Comment (okuta) [inline cupy/core/core.pyx]: Please remove this line.

Comment (unnonouno) [inline cupy/core/core.pyx]: Why?

Comment (okuta) [inline cupy/core/core.pyx]: For flake8.
`./cupy/core/core.pyx:372:5: E303 too many blank lines (2)`

File: cupy/core/core.pyx [hunk 2]
```diff
@@ -342,38 +342,31 @@ cdef class ndarray:
...
+            return self.astype(self.dtype, copy=True, order=order)
 
-        a = self
-        if order == 'C' and not self._c_contiguous:
-            with self.device:
-                a = ascontiguousarray(self)
-            if a.data.device.id == device.get_device_id():
-                return a
-        elif order == 'F' and not self._f_contiguous:
+        if device.Device() == self.device:
```
Comment (okuta) [inline cupy/core/core.pyx]: Please use `if self.data.device is None or self.data.device.id == device.get_device_id():` for performance.

**Comment by unnonouno** [comment]
I updated it.

**Comment by okuta** [comment]
LGTM!

**Pull request merged**